src/test/java/BaseTest.java

```diff
@@ -1,11 +1,90 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
 import org.testng.annotations.BeforeSuite;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Parameters;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.time.Duration;
 
 public class BaseTest {
 
+    public static WebDriver driver = null;
+    public static Actions actions = null;
+
+    public static WebDriverWait wait = null;
+
+    public static String url = null;
+    //   public static String url = "https://qa.koel.app/";
+
     @BeforeSuite
     static void setupClass() {
+
         WebDriverManager.chromedriver().setup();
     }
+
+    @BeforeMethod
+    @Parameters({"BaseURL"})
+    public void launchBrowser(String BaseURL) {
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+        options.addArguments("--disable-notifications", "--remote-allow-origins=*", "--incognito", "--start-maximized");
+        options.setExperimentalOption("excludeSwitches", new String[]{"enable-automation"});
+
+        driver = new ChromeDriver(options);
+        driver = pickBrowser(System.getProperty("browser"));
```

Comment:
Hello,
Your homework 24 has been reviewed, but the script failed due to [error ](error: unreported exception MalformedURLException; must be caught or declared to be thrown)
// In the pickBrowser function you use throws MalformedURLException, so when you calling pickBrowser func into launchBrowser function, you should also add throws MalformedURLException for launchBrowser function
![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/144403954/3c77d15d-8442-4a78-a7ee-fd3c79750cba)